extensions/guacamole-auth-quickconnect/src/main/java/org/apache/guacamole/auth/quickconnect/QuickConnectionGroup.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.guacamole.auth.quickconnect;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.apache.guacamole.GuacamoleException;
+import org.apache.guacamole.GuacamoleSecurityException;
+import org.apache.guacamole.net.GuacamoleTunnel;
+import org.apache.guacamole.net.auth.AbstractConnectionGroup;
+import org.apache.guacamole.net.auth.ConnectionGroup;
+import org.apache.guacamole.protocol.GuacamoleClientInformation;
+
+/**
+ * Provides a simple, single-level connection group used for
+ * temporarily storing the Connection objects created by users.
+ */
+public class QuickConnectionGroup extends AbstractConnectionGroup {
+
+    /**
+     * A set that will store the Connection identifiers for this group.
+     */
+    private Set<String> connectionIdentifiers =
+            new HashSet<String>(Collections.<String>emptyList());
+
+    /**
+     * Set up a QuickConnectionGroup with the provided name and
+     * identifier.
+     *
+     * @param name
+     *     The name of the QuickConnectionGroup.
+     *
+     * @param identifier
+     *     The identifier of the QuickConnectionGroup.
+     */
+    public QuickConnectionGroup(String name, String identifier) {
+
+        setName(name);
+        setIdentifier(identifier);
+        setType(ConnectionGroup.Type.ORGANIZATIONAL);
```

Comment:
Though older code within Guacamole does this, it's bad practice to invoke overridable functions of the class being instantiated within its own constructor. If the setters belong to the superclass, `super` should be used explicitly. Some IDEs will warn of this.

https://stackoverflow.com/questions/3404301/whats-wrong-with-overridable-method-calls-in-constructors

Reply:
Okay, but if the superclass doesn't have a constructor that calls these, how should I handle that?  Override the individual methods inside the class itself and then call them?  I'm looking at `AbstractConnectionGroup` and `ConnectionGroup` and `Identifiable` and not seeing a constructor that actually takes these arguments and initializes the various objects?

Reply:
We probably should provide a constructor, but you can get around this for the time being by using `super.setName()`, `super.setIdentifier()`, etc., as that will invoke a function against the superclass which is definitely initialized at this point.

Reply:
Done.